include/ensmallen_bits/cmaes/pop_cmaes_impl.hpp

```diff
@@ -0,0 +1,168 @@
+/**
+ * @file ipop_cmaes_impl.hpp
+ * @author Marcus Edel
+ * @author Benjami Parellada
+ *
+ * Implementation of the IPOP Covariance Matrix Adaptation Evolution Strategy
+ * as proposed by A. Auger and N. Hansen in "A Restart CMA Evolution
+ * Strategy With Increasing Population Size" and BIPOP Covariance Matrix
+ * Adaptation Evolution Strategy as proposed by N. Hansen in "Benchmarking 
+ * a BI-population CMA-ES on the BBOB-2009 function testbed".
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_CMAES_POP_CMAES_IMPL_HPP
+#define ENSMALLEN_CMAES_POP_CMAES_IMPL_HPP
+
+#include "pop_cmaes.hpp"
+#include <ensmallen_bits/function.hpp>
+#include <random>
```

Comment:
```suggestion
```

We can actually use Armadillo's RNGs directly in the class, and avoid the inclusion of random.  See the comments below.